nowradio/refresh_locations.py

```diff
@@ -0,0 +1,63 @@
+from dbManager import Queries, get_connection
+import ourUtils
+
+# This batch is used to fill in the longitude and latitude data
+# for every ip in our station_info table. It collects all of our ips
+# and spawns off threads which use a geolocation API (ourUtils.location_from_ip)
+# to find the lat and long values for each ip. 
+# Finally inserts these values in the appropriate rows.
+
+
+def insert_locations_by_ip(conn, location_tuples):
+    for location in location_tuples:
+        data = {
+            'ip': location[0],
+            'latitude': location[1][0],
+            'longitude': location[1][1]
+        }
+        ourUtils.db_quick_query(
+            conn,
+            Queries.insert_location_for_ip,
+            data
+        )
+
+
+def get_location_tuples(ip_list):
+    return [(ip, ourUtils.location_from_ip(ip.split(':')[0])) for ip in ip_list]
+
+
+def get_and_insert_locations(ip_list, conn):
+    location_tuples = get_location_tuples(ip_list)
+    insert_locations_by_ip(conn, location_tuples)
+
+
+def get_ip_list(conn):
+    # Returns a list of ips.
+    count = ourUtils.db_quick_query(conn, Queries.get_total_rows)
+    total_rows = count[0][0]
+    batch_size = 500
+    offset = 0
+    ip_list = []
+    while offset < total_rows:
```

Comment:
I'm not sure that batching will help prevent running out of memory, because ip_list will always end up having 10,000 entries. And I don't think that reducing table lock is worth the code complexity, especially at our current scale. 
If we wanted to start doing batch queries, I would want to make one utility function, and change all large queries to use it. But at this point I don't think it is worth it.